### PR TITLE
fix: update GoogleUtils#getVersion() to use stable logic

### LINKED ISSUE #1451: Android/Java 1.30.6 causing crash due to PR #1419
Prompted by this SO post with more details: https://stackoverflow.com/questions/59243821/youtube-api-exception-in-initializer-error/59298544#59298544

#### Environment details

- OS: Android
- Java version: 8
- google-api-java-client version: 1.30.6

#### Steps to reproduce

1. Try to access GoogleUtils.VERSION with build.gradle looking like:
`dependencies {
  implementation ('com.google.api-client:google-api-client:1.30.6')
  implementation ('com.google.api-client:google-api-client-android:1.30.6')
  implementation ('com.google.apis:google-api-services-drive:+')
}` 

#### Stacktrace
```
Caused by: java.lang.IllegalStateException: No successful match so far
        at java.util.regex.Matcher.ensureMatch(Matcher.java:1116)
        at java.util.regex.Matcher.group(Matcher.java:382)
        at com.google.api.client.googleapis.GoogleUtils.<clinit>(Unknown Source:26)
```

Please let me know if this is a problem with our own setups, as far as I can tell there's not a whole lot I can do.

Thanks!

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1451

### CODE REVIEW

File: google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java [hunk 1]
```diff
@@ -89,20 +89,18 @@ public static synchronized KeyStore getCertificateTrustStore()
...
+    String version = "unknown-version";
+    try (InputStream inputStream =
+        GoogleUtils.class.getResourceAsStream("google-api-client.properties")) {
+      if (inputStream != null) {
+        final Properties properties = new Properties();
+        properties.load(inputStream);
+        version = properties.getProperty("google-api-client.version");
       }
+    } catch (IOException e) {
+      // ignore
```
Comment (elharo) [inline google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java]: do we really want to ignore this and not fill in a default?

File: google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java [hunk 2]
```diff
@@ -89,20 +89,18 @@ public static synchronized KeyStore getCertificateTrustStore()
...
-        }
-      } catch (IOException e) {
-        // ignore
+    // attempt to read the library's version from a properties file generated during the build
+    // this value should be read and cached for later use
+    String version = "unknown-version";
+    try (InputStream inputStream =
+        GoogleUtils.class.getResourceAsStream("google-api-client.properties")) {
+      if (inputStream != null) {
+        final Properties properties = new Properties();
```
Comment (elharo) [inline google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java]: probably doesn't need to be final

File: google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java (hunk 1 above)
Comment (elharo) [reply google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java]: NVM, that's what we do effectively.

File: google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java [hunk 3]
```diff
@@ -89,20 +89,18 @@ public static synchronized KeyStore getCertificateTrustStore()
...
+        properties.load(inputStream);
+        version = properties.getProperty("google-api-client.version");
```
Comment (elharo) [inline google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java]: this could be null. You should supply a default here.

Comment (chingor13) [reply google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java]: Done. Moved the default to the return statement while checking for null.

Comment (elharo) [reply google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java]: Or you could write

```
version = properties.getProperty("google-api-client.version". "unknown-version");
```

up to you.

**Comment by Samikay** [comment]
Did this make it into 1.30.7? Cause it's still broken.

     Caused by: java.lang.IllegalStateException: No successful match so far
        at java.util.regex.Matcher.ensureMatch(Matcher.java:1116)
        at java.util.regex.Matcher.group(Matcher.java:382)
        at c.c.b.a.c.a.<clinit>(:66)
        at com.google.api.services.drive.Drive.<clinit>(:44) 
        at com.google.api.services.drive.Drive$Builder.build(:9813)

**Comment by elharo** [comment]
related #1467

**Pull request merged**